opentelemetry-sdk/src/metrics/internal/mod.rs

```diff
@@ -135,6 +134,66 @@ where
             );
         }
     }
+
+    /// Iterate through all attribute sets and populate `DataPoints` in readonly mode.
```

Comment:
nit: Add a comment here to state that this is used in Cumulative temporality mode where ValueMap is not cleared, and hence Read lock is sufficient.